virtualization-and-containers/material/docker_slides.md

```diff
@@ -0,0 +1,252 @@
+  .reveal strong {
+  font-weight: bold;
+    color: orange;
+  }
+  .reveal p {
+    text-align: left;
+  }
+  .reveal section h1 {
+    color: orange;
+  }
+  .reveal section h2 {
+    color: orange;
+  }
+</style>
+
+# Docker
+
+<img src="https://www.docker.com/sites/default/files/d8/2019-07/Moby-logo.png" width=20%; style="margin-left:auto; margin-right:auto; padding-top: 25px; padding-bottom: 25px">
+
+[https://www.docker.com/company/newsroom/media-resources](https://www.docker.com/company/newsroom/media-resources)
+
+---
+
+## What is Docker? 1/2
+
+- Docker, Inc.
+  - Company promoting Docker
+- Docker Desktop
+  > Developer productivity tools and a local Kubernetes environment.
+- Docker Engine
+  > Docker Engine is an open source containerization technology for building and containerizing your applications.
+
+---
+
+## What is Docker? 2/2
+
+- Docker Hub
+  > Cloud-based application registry and development team collaboration services.
+- Docker Compose
+  > Compose is a tool for defining and running multi-container Docker applications.
+- According to documentation
+  > Docker is an open platform for developing, shipping, and running applications.
+- We use definition from documentation.
+
+---
+
+## Typical Docker Applications
+
+- Applications as Microservices
+- Containers for consistent development environment
+- Containers for consistent testing environment
+- Portable format for sharing applications
+- Avoid tedious installation procedures by providing Docker container ([FEniCS](https://fenicsproject.org/download/), [GitLab](https://docs.gitlab.com/ee/install/docker.html)...)
+
+---
+
+## Building Blocks 1/2
+
+- Docker daemon `dockerd`
+  - Controlling instance of containers and reacts to API requests
+  - Server process
+- Docker client
+  - User interface/tools to interact with, create, manage containers etc. via daemon
+- Docker registries
+  - Registries that manage Docker images to be used
+
+---
+
+## Building Blocks 2/2
+
+- Docker objects
+  - **Images**
+    - Read-only template for creating a container
+    - An image can be based on another image
+  - **Containers**
+    - Runnable instance of an image
+
+---
+
+## Docker Architecture
+
+<img src="https://docs.docker.com/engine/images/architecture.svg" width=80%; style="margin-left:auto; margin-right:auto; padding-top: 25px; padding-bottom: 25px">
+
+[https://docs.docker.com/engine/images/architecture.svg](https://docs.docker.com/engine/images/architecture.svg)
+
+---
+
+## Connection to Host
+
+- Container communicates via daemon `dockerd` (runs as   root)
+- Strong isolation (`namespaces` and `cgroups`)
+  - You cannot access Host filesystem by default
+  - Several [mount options](https://docs.docker.com/storage) available
+
+---
+
+## Requirements
+
+- Root rights for installation
+- `dockerd` runs as root -> Interaction needs root rights
+  - Prefix commands with `sudo`
+  - Be member of group `docker` (=makes you root), expected by some applications (e.g. `act`)
```

Comment:
Quickly mention what `act` is?
e.g. that we might come back to this in later lectures?

Reply:
I will mention this. It is also in the notes in `overview.md`.